### DISCUSSION TITLE
Any View API to get row number by an index value

### DISCUSSION THREAD

**Discussion by nekw** [perspective-dev/perspective#1775] [Q&A] [answered]
Is there any API or quick way to get row number providing a index value from a View

E.g.
{
Field1: [k1, k2, k3, k4],
Field2: [a, b, c, d]
}
Where Field1 is index

View.GetRowNumberByIndexValue('k2') 

Return 2

**Comment by texodus** [comment, marked as answer]
You can use a filter:

```javascript
await table.view({filter: [["Field1", "==", "k2"]]});
```

**Reply by nekw** [reply]
How to get the value 2 from this? (2 means 2nd row here)

**Reply by texodus** [reply]
```javascript
(await view.to_json())[0]
```

You may want to review the docs [retrieving data from a `View`](https://perspective.finos.org/docs/md/js.html#serializing-data-using-to_) as well as [the `View` constructor transform with interactive examples](https://perspective.finos.org/docs/md/view.html).  Beyond just answering your immediate questions, there are some important details there that are specific to wasm (distinct from JavaScript), e.g. [memory management](https://perspective.finos.org/docs/md/js.html#deleting-a-table-or-view).
